Shared/UI/Face Detection/FaceDetectionViewController.swift

```diff
@@ -92,31 +88,128 @@ class FaceDetectionViewController: ImageCaptureViewController {
     }
 
     override func photoOutput(_ output: AVCapturePhotoOutput,
-                     didFinishProcessingPhoto photo: AVCapturePhoto,
-                     error: Error?) {
+                              didFinishProcessingPhoto photo: AVCapturePhoto,
+                              error: Error?) {
 
         guard let connection = output.connection(with: .video) else { return }
         connection.automaticallyAdjustsVideoMirroring = true
 
         guard let ciImage = self.currentCIImage else { return }
 
-        let image = UIImage(ciImage: ciImage, scale: 1.0, orientation: .up)
+        // If we find a face in the image, we'll crop around it and store it here.
+        var finalCIImage = ciImage
 
-        let imageOptions =  NSMutableDictionary(object: NSNumber(value: 5) as NSNumber, forKey: CIDetectorImageOrientation as NSString)
+        let imageOptions = NSMutableDictionary(object: NSNumber(value: 5) as NSNumber,
+                                               forKey: CIDetectorImageOrientation as NSString)
         imageOptions[CIDetectorEyeBlink] = true
-        let accuracy = [CIDetectorAccuracy: CIDetectorAccuracyHigh]
+        let accuracy = [CIDetectorAccuracy : CIDetectorAccuracyHigh]
         let faceDetector = CIDetector(ofType: CIDetectorTypeFace, context: nil, options: accuracy)
         let faces = faceDetector?.features(in: ciImage, options: imageOptions as? [String : AnyObject])
 
         if let face = faces?.first as? CIFaceFeature {
             self.eyesAreClosed = face.leftEyeClosed && face.rightEyeClosed
             self.isSmiling = face.hasSmile
+
+            var adjustedFaceBounds = face.bounds
+            adjustedFaceBounds.size.height = face.bounds.height * 1.8
+            adjustedFaceBounds.size.width = adjustedFaceBounds.height
+            adjustedFaceBounds.centerY = face.bounds.centerY + face.bounds.height * 0.2
+            adjustedFaceBounds.centerX = face.bounds.centerX
+
+            finalCIImage = ciImage.cropped(to: adjustedFaceBounds)
         } else {
             self.eyesAreClosed = false
-            self.isSmiling = false 
+            self.isSmiling = false
         }
 
+        // CGImages play nicer with UIKit.
+        // Per the docs: "Due to Core Image's coordinate system mismatch with UIKit, this filtering
+        // approach may yield unexpected results when displayed in a UIImageView with contentMode."
+        let context = CIContext()
+        let cgImage = context.createCGImage(finalCIImage, from: finalCIImage.extent)!
 
+        let image = UIImage(cgImage: cgImage, scale: 1, orientation: .up)
```

Comment:
It's annoying, but ImageViews have trouble displaying CI images properly. Converting to a CG image.